### PR TITLE
"Add About and Contact components with routing and styling"

### PR DESCRIPTION
**Pull request by Abhinav8639**
### Related Issue
- Closes: #6 

---

### Description

 Added About.jsx and Contact.jsx components inside the components folder.
Implemented routing for the About and Contact pages in App.jsx.
Created about.html and contact.html files to render the components when clicked.
Added necessary import statements in App.jsx and other relevant files.
Designed CSS for the About and Contact components to match the app's styling.
Updated the navigation in App.jsx to include the new About and Contact components.
Ensured responsiveness of the About and Contact sections for both desktop and mobile views. 

---

### How Has This Been Tested?

Tested by running the app and ensuring that clicking on the About and Contact links correctly navigates to their respective components.
Verified that the pages are styled correctly and are responsive on various screen sizes.
Checked that the routes work correctly in the app’s navigation.

---

### Screenshots (if applicable)

Screenshot (5632)
Screenshot (5633)

Screenshot (5634)

---

### Type of Change

- [ ] Bug fix
- [ v] New feature
- [ ] Code style update
- [ ] Breaking change
- [ ] Documentation update

### CODE REVIEW

**Comment by mehul-m-prajapati** [comment]
@Abhinav8639 : Remove html pages from public folder. It should be there in pages directory only

File: public/about.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>About</title>
```
Comment (yashksaini-coder) [inline public/about.html:4]: Remove this duplicate about html page @Abhinav8639

File: src/pages/Home/About.css [hunk 2]
```diff
@@ -0,0 +1,68 @@
+.about-container {
+    padding: 40px;
+    max-width: 900px;
+    margin: 0 auto;
+    font-family: Arial, sans-serif;
+  }
+  
+  .hero {
```
Comment (yashksaini-coder) [inline src/pages/Home/About.css:8]: Remove the css file @Abhinav8639

File: src/pages/Home/Contact.css [hunk 3]
```diff
@@ -0,0 +1,66 @@
+.contact-container {
+    padding: 40px;
+    max-width: 900px;
+    margin: 0 auto;
+    font-family: Arial, sans-serif;
+  }
+  
+  .contact-methods {
```
Comment (yashksaini-coder) [inline src/pages/Home/Contact.css:8]: Remove this css file @Abhinav8639

File: app.js [hunk 4]
```diff
@@ -25,13 +25,19 @@ async function getGithubData(url) {
...
 
+// Route to fetch GitHub data
 app.post('/fetch', async (req, res) => {
-    const username = req.body.username;
-    const url = `/${username}`;
+    const { username } = req.body;
+    if (!username) {
+        return res.status(400).send('Username is required');
+    }
+
```
Comment (yashksaini-coder) [inline app.js:39]: Don't update this file, remove the changes from this

**Review by yashksaini-coder — CHANGES_REQUESTED** [review]
@Abhinav8639 please complete the requested changes, and make sure to add proper issue number next time, in the PR description.

File: app.js (hunk 4 above)
Comment (Abhinav8639) [reply app.js:39]: hello, you mentioned not to modify the app.js file. I wanted to clarify how you'd like me to proceed with implementing the routing or features for the Contact and About pages. Should I create a new file for the routes or features, or should they be handled in a different way?

Comment (mehul-m-prajapati) [reply app.js:39]: @Abhinav8639 : app.js was an old implementation. I have removed it. Routing is done App.jsx only. Your changes are good in that file

**Comment by ZeDD0405** [comment]
Related Issue
- Closes: Add About Us, Contact Us pages #6

Description
- Created about.html and contact.html pages with responsive designs.
- Designed and implemented hover and shadow effects for navigation links and section headings.
- Ensured navigation between the About Us and Contact Us pages works seamlessly.
- Added CSS styling for the About and Contact sections to match the overall theme and design.
- Included form elements in the Contact Us page with interactive features such as hover effects on form fields and buttons.
- Tested for responsiveness on both desktop and mobile views, ensuring a consistent user experience across different devices.

How Has This Been Tested?
- Verified functionality by opening the pages in the browser and testing navigation between pages.
- Checked hover effects, shadow effects, and form interactions on both About and Contact pages.
- Ensured the design is responsive by resizing the browser window to confirm proper layout adjustments on various screen 
   
Vite + React - Google Chrome 16-11-2024 10_31_58
Vite + React - Google Chrome 16-11-2024 10_32_01
Vite + React - Google Chrome 16-11-2024 10_32_04
sizes.

File: app.js [hunk 5]
```diff
@@ -1,69 +1,69 @@
...
-
-// Start the server
-app.listen(PORT, () => {
-    console.log(`Server is running on http://localhost:${PORT}`);
-});
+const express = require('express');
+const gs = require('github-scraper');
+
+const app = express();
+const PORT = 3000;
```
Comment (yashksaini-coder) [inline app.js:5]: As Mehul sir said previously, remove this file. We don't need this file.

File: src/pages/About/About.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        </p>
+      </section>
+
+      {/* Features Section */}
+      <section className="features bg-gray-100 py-12">
+        <h2 className="text-3xl font-semibold text-center">What We Do</h2>
+        <div className="flex justify-around mt-8">
+          <div className="feature-item text-center">
+            <div className="feature-icon text-4xl">🔍</div>
+            <h3 className="font-semibold mt-4">Simple Issue Tracking</h3>
```
Comment (yashksaini-coder) [inline src/pages/About/About.jsx:26]: @mehul-m-prajapati sir is it okay to store this About.jsx page in another dir just for this single page. 

Same goes for Contact.jsx page

File: src/App.jsx [hunk 7]
```diff
@@ -1,29 +1,48 @@
...
+import About from './pages/About/About';  // Import the About component
+import Contact from './pages/Contact/Contact';  // Import the Contact component
+=======
 import { Routes, Route, Navigate } from 'react-router-dom';
 import Navbar from './components/Navbar';
 import Footer from './components/Footer';
 import Home from './pages/Home/Home';
 import About from './pages/Home/About'; // Adjust the path if necessary
 import Contact from './pages/Home/Contact'; // Adjust the path if necessary
+>>>>>>> da11d282ff2b82196d95284949aed1f8fa36f56a
```
Comment (yashksaini-coder) [inline src/App.jsx:15]: Why is this showing here @Abhinav8639

File: src/components/Navbar.jsx [hunk 8]
```diff
@@ -6,13 +6,21 @@ function Navbar() {
...
         {/* Brand Name */}
         <div className="text-2xl font-bold hover:text-gray-300 cursor-pointer">
+<<<<<<< HEAD
+          GitHub Tracker
+=======
           <Link to="/">GitHub Tracker</Link>
+>>>>>>> da11d282ff2b82196d95284949aed1f8fa36f56a
         </div>
 
         {/* Links/Buttons */}
```
Comment (yashksaini-coder) [inline src/components/Navbar.jsx:16]: The file is wrong please review your code, there are merge headers in the file

**Review by yashksaini-coder — CHANGES_REQUESTED** [review]
@mehul-m-prajapati there are still many things that are not correct, please fix them and raise a new PR properly.

File: src/components/Navbar.jsx (hunk 8 above)
Comment (Abhinav8639) [reply src/components/Navbar.jsx:12]: something gone wrong and files got replaced i will correct it

**Pull request closed without merging**